tests/unit/rest.py

```diff
@@ -231,51 +231,58 @@ def test_none_return(self):
 
 @pytest.fixture()
 def validation_setup():
-    # There's a mix of PUT and POST in these; mixing it up may give us
-    # better coverage. The particulars of which calls get which methods are
-    # arbitrary.
+    # There's a mix of GET, PUT and POST in these; mixing it up may give us
+    # better coverage. The particulars of which PUT and POST calls get which
+    # methods are arbitrary, though GET calls have some specificity.
 
     # We have four kinds of calls we want to validate here:
 
     # 1. No arguments in the URL or body (no_args)
-    @rest.rest_call('POST', '/no/args', Schema({}))
+    @rest.rest_call(['GET','POST'], '/no/args', Schema({}))
     def no_args():
         pass
 
     # 2. Argument in the URL and not in the body (url_args)
-    @rest.rest_call('POST', '/url/args/<arg1>/<arg2>', Schema({
+    @rest.rest_call(['GET','POST'], '/url/args/<arg1>/<arg2>', Schema({
         'arg1': basestring,
         'arg2': basestring,
     }))
     def url_args(arg1, arg2):
         return json.dumps([arg1, arg2])
 
     # 3. Arguments in both the URL and body (mixed_args)
-    @rest.rest_call('PUT', '/mixed/args/<arg1>', Schema({
+    @rest.rest_call(['GET','PUT'], '/mixed/args/<arg1>', Schema({
         'arg1': basestring,
         'arg2': basestring,
     }))
     def mixed_args(arg1, arg2):
         return json.dumps([arg1, arg2])
 
-    # 4. Arguments in body and not in the URL.
-    @rest.rest_call('POST', '/body/args', Schema({
+    # 4. Arguments in body (query parameters for GET) and not in the URL.
+    @rest.rest_call(['GET','POST'], '/just/args', Schema({
         'arg1': basestring,
         'arg2': basestring,
     }))
-    def body_args(arg1, arg2):
+    def just_args(arg1, arg2):
+        return json.dumps([arg1, arg2])
+
+    # 5. One optional argument.
+    @rest.rest_call(['GET', 'POST'], '/one/optional', Schema({
+        'arg1': basestring,
+        Optional('arg2'): Use(int),
+    }))
+    def one_optional(arg1, arg2=-42):
```

Comment:
This is another one that's hard to understand from the call site. Things that aren't obvious from that perspective:

1. *which* argument is optional?
2. There's no indiciation that there are any other requirements (i.e. it's not obvious you need to pass it something that is convertable to an int).